src/esp/geo/VoxelGrid.cpp

```diff
@@ -0,0 +1,877 @@
+        if (hit) {
+          setVoxel(Mn::Vector3i(i, j, ind), "posZShadow", true);
+        } else if (getVoxel<bool>(Mn::Vector3i(i, j, ind), "Boundary")) {
+          hit = true;
+          setVoxel(Mn::Vector3i(i, j, ind), "posZShadow", true);
+        }
+      }
+    }
+  }
+
+  // create int grid
+  addGrid<int>(gridName);
+  bool nX = false, pX = false, nY = false, pY = false, nZ = false, pZ = false;
+  int hash = 0;
+  // fill in int grid with voting approach
+  for (int i = 0; i < m_voxelGridDimensions[0]; i++) {
+    for (int j = 0; j < m_voxelGridDimensions[1]; j++) {
+      for (int k = 0; k < m_voxelGridDimensions[2]; k++) {
+        Mn::Vector3i index = Mn::Vector3i(i, j, k);
+        if (getVoxel<bool>(index, "Boundary")) {
+          setVoxel(index, gridName, 0);
+          continue;
+        }
+        nX = !getVoxel<bool>(index, "negXShadow");
+        pX = !getVoxel<bool>(index, "posXShadow");
+        nY = !getVoxel<bool>(index, "negYShadow");
+        pY = !getVoxel<bool>(index, "posYShadow");
+        nZ = !getVoxel<bool>(index, "negZShadow");
+        pZ = !getVoxel<bool>(index, "posZShadow");
+        // || ((nX || pX) && (nY || pY) && (nZ || pZ))
+        if (((nX && pX) || (nY && pY) || (nZ && pZ)) ||
+            ((nX || pX) && (nY || pY) && (nZ || pZ))) {
+          // Exterior (+inf)
+          setVoxel(index, gridName, INT_MAX);
+        } else {
+          // Interior (-inf)
+          setVoxel(index, gridName, INT_MIN);
+        }
+      }
+    }
+  }
+  // remove the transient utility grids
+  removeGrid("negXShadow");
+  removeGrid("posXShadow");
+  removeGrid("negYShadow");
+  removeGrid("posYShadow");
+  removeGrid("negZShadow");
+  removeGrid("posZShadow");
+}
+
+// Manhattan distance SDF - starting from the interior exterior voxel grid,
+// computes SDF in terms of manhattan distance with double sweep approach
+void VoxelGrid::generateManhattanDistanceSDF(const std::string& gridName) {
+  // check to see if Interior/Exterior grid exists, if not, generate it
+  if (grids_.find("InteriorExterior") == grids_.end()) {
+    generateInteriorExteriorVoxelGrid();
+  }
+  // create new intGrid and copy data from interior/exterior grid
+  addGrid<int>(gridName);
+  auto intExtGrid = getGrid<int>("InteriorExterior");
+  auto sdfGrid = getGrid<int>(gridName);
+
+  for (int i = 0; i < m_voxelGridDimensions[0]; i++) {
+    for (int j = 0; j < m_voxelGridDimensions[1]; j++) {
+      for (int k = 0; k < m_voxelGridDimensions[2]; k++) {
+        // copy value
+        sdfGrid[i][j][k] = intExtGrid[i][j][k];
+      }
+    }
+  }
+
+  // 1st sweep
+  for (int i = 0; i < m_voxelGridDimensions[0]; i++) {
+    for (int j = 0; j < m_voxelGridDimensions[1]; j++) {
+      for (int k = 0; k < m_voxelGridDimensions[2]; k++) {
+        int i_behind = INT_MAX, j_behind = INT_MAX, k_behind = INT_MAX;
+        if (isValidIndex(Mn::Vector3i(i - 1, j, k))) {
+          i_behind = abs(std::max(sdfGrid[i - 1][j][k], -2147483646));
```

Comment:
```suggestion
          i_behind = abs(std::max(sdfGrid[i - 1][j][k], -INT_MAX));
```

? :)

Same in all the occurences below.